packages/react/src/primitives/Select/__tests__/Select.test.tsx

```diff
@@ -0,0 +1,203 @@
+    render(
+      <Select
+        testId={testId}
+        name={name}
+        size={size}
+        variation={variation}
+        isDisabled={false}
+        isRequired={false}
+      >
+        <option value="1">1</option>
+        <option value="2">2</option>
+        <option value="3">3</option>
+      </Select>
+    );
+
+    const select = (await screen.findByTestId(
+      'test-select'
+    )) as HTMLSelectElement;
+    expect(select.nodeName).toBe('SELECT');
+    expect(select.childNodes).toHaveLength(3);
+    expect(select).toHaveAttribute('name', name);
+    expect(select).toHaveAttribute('data-size', size);
+    expect(select).toHaveAttribute('data-variation', variation);
+    expect(select).toHaveAttribute('class', ComponentClassNames.Select);
+    expect(select).not.toBeDisabled();
+    expect(select).not.toBeRequired();
+  });
+
+  it('should render placeholder correctly if it is set', async () => {
+    render(
+      <Select placeholder={placeholder} defaultValue="">
+        <option value="1">1</option>
+        <option value="2">2</option>
+        <option value="3">3</option>
+      </Select>
+    );
+
+    const placeholderElement = (await screen.findByText(
+      placeholder
+    )) as HTMLOptionElement;
+    expect(placeholderElement.nodeName).toBe('OPTION');
+    expect(placeholderElement).toHaveAttribute('value', '');
+    expect(placeholderElement).toBeDisabled();
+    expect(placeholderElement.selected).toBeTruthy();
+  });
+
+  it('should be able to select an option if not be disabled', async () => {
+    render(
+      <Select testId={testId} placeholder={placeholder} defaultValue="">
+        <option value="1">1</option>
+        <option value="2">2</option>
+        <option value="3">3</option>
+      </Select>
+    );
+    const select = (await screen.findByTestId(
+      'test-select'
+    )) as HTMLSelectElement;
+    const placeholderOption = screen.getByRole('option', {
+      name: placeholder,
+    }) as HTMLOptionElement;
+    const optionOne = screen.getByRole('option', {
+      name: '1',
+    }) as HTMLOptionElement;
+    const optionTwo = screen.getByRole('option', {
+      name: '2',
+    }) as HTMLOptionElement;
+    const optionThree = screen.getByRole('option', {
+      name: '3',
+    }) as HTMLOptionElement;
+
+    expect(placeholderOption.selected).toBeTruthy();
+
+    userEvent.selectOptions(select, '1');
+    expect(optionOne.selected).toBeTruthy();
+    expect(optionTwo.selected).toBeFalsy();
+    expect(optionThree.selected).toBeFalsy();
+
+    userEvent.selectOptions(select, '2');
+    expect(optionOne.selected).toBeFalsy();
+    expect(optionTwo.selected).toBeTruthy();
+    expect(optionThree.selected).toBeFalsy();
+
+    userEvent.selectOptions(select, '3');
+    expect(optionOne.selected).toBeFalsy();
+    expect(optionTwo.selected).toBeFalsy();
+    expect(optionThree.selected).toBeTruthy();
+  });
+
+  it('should not be able to select an option if disabled', async () => {
```

Comment:
Alternatively you could just test that the the disabled attribute is passed through correctly to the select element. We probably don't need to test this, since disabled selects not being interactive is default behavior of the browser, not functionality we are adding.

Reply:
For sure, let me trace back and remove those browser default behavior tests in favor of controlled component tests